### PR TITLE
ENT-5035: schema.sql breakage should fail faster during install

### PR DESCRIPTION
**Pull request by craigcomstock**

### CODE REVIEW

**Comment by craigcomstock** [comment]
This change made the install failure very obvious what the problem was, instead of passing along and showing a much less obvious reason later.

```
+ su cfpostgres -c /var/cfengine/bin/psql cfmp
CREATE ROLE
+ chown cfpostgres /var/cfengine/share/db/ootb_import.sql /var/cfengine/share/db/ootb_settings.sql /var/cfengine/share/db/schema.sql /var/cfengine/share/db/schema_settings.sql
+ cd /tmp
+ chown cfpostgres /var/cfengine/share/db/schema.sql
+ su cfpostgres -c /var/cfengine/bin/psql cfdb -v "ON_ERROR_STOP=1" -f /var/cfengine/share/db/schema.sql
psql:/var/cfengine/share/db/schema.sql:32: ERROR:  syntax error at or near "BROKEN"
LINE 1: BROKEN
        ^
+echo '=== END /var/log/CFEngine*-Install.log ==='
=== END /var/log/CFEngine*-Install.log ===
+return 1
The command "buildscripts/build-scripts/travis" exited with 1.
```

less obvious reason before this change:

```
+ /var/cfengine/httpd/php/bin/php /var/cfengine/httpd/htdocs/index.php cli_tasks inventory_refresh
Error:: A Database Error Occurred
Error Number: 42P01/7,ERROR:  relation "v_inventory" does not exist
LINE 1: INSERT INTO __inventory (SELECT * FROM v_inventory WHERE val...
                                               ^
QUERY:  INSERT INTO __inventory (SELECT * FROM v_inventory WHERE values IS NOT NULL)
CONTEXT:  PL/pgSQL function update_inventory() line 12 at SQL statement,SELECT * FROM update_inventory(),Filename: core/Cf_DB_pdo_driver.php,Line Number: 93+echo '=== END /var/log/CFEngine*-Install.log ==='
=== END /var/log/CFEngine*-Install.log ===
+return 1
The command "buildscripts/build-scripts/travis" exited with 1.
```

File: packaging/common/cfengine-hub/postinstall.sh [hunk 1]
```diff
@@ -783,19 +783,19 @@ else
   # Ensure cfpostgres can read the sql files it will import. And that they are
   # restored to restrictive state after import ENT-2684
   chown cfpostgres $PREFIX/share/db/*.sql
-  (cd /tmp && chown cfpostgres "$PREFIX/share/db/schema.sql" && su cfpostgres -c "$PREFIX/bin/psql cfdb -f $PREFIX/share/db/schema.sql" && chown root "$PREFIX/share/db/schema.sql")
+  (cd /tmp && chown cfpostgres "$PREFIX/share/db/schema.sql" && su cfpostgres -c "$PREFIX/bin/psql cfdb -v \"ON_ERROR_STOP=1\" -f $PREFIX/share/db/schema.sql" && chown root "$PREFIX/share/db/schema.sql")
```
Comment (vpodzime) [inline packaging/common/cfengine-hub/postinstall.sh]: I don't think the double-quotes are needed so I'd prefer removing them than bothering escaping them.

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2911/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2911/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2911/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins again please, last time it was checked_outputs failing which seems unrelated to this change.

**Comment by craigcomstock** [comment]
Current travis failure is as expected. Will check on latest jenkins build to ensure fail-fast error on broken schema.sql. If good, then will remove the broken nova PR and re-run to ensure everything passes (travis and jenkins) :+1:

**Comment by craigcomstock** [comment]
@cf-bottom jenkins should fail in the right way now I think.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/3023/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/3023/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-3023/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins again please, this time hopefully canary will fail fast during reporting tests with change in related nova PR.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/3024/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/3024/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-3024/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins again, this time maybe verbose sql logs on failure.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/3025/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/3025/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-3025/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins again, thanks.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/3026/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/3026/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-3026/

**Comment by craigcomstock** [comment]
@cf-bottom I have removed the breakage in schema.sql so another jenkins please, this one should be green.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/3050/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/3050/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-3050/

**Comment by craigcomstock** [comment]
@cf-bottom let's try jenkins again, this time without the associated nova PR. The nova PR was trying to add in more schema testing to ensure that schema problems are seen faster/better in the builds. But for now, let's see if this change works on it's own.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4033/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4033/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4033/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4043/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4043/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4043/

**Comment by craigcomstock** [comment]
Failed on tests involving debian 7.

```
    tests/all_packages/test_restarting_hub_agents.py.PACKAGES_HUB_x86_64_linux_debian_7,ubuntu-12-04-x64
    tests/all_packages/test_restarting_hub_agents.py.PACKAGES_HUB_x86_64_linux_debian_7,ubuntu-14-04-x64
```

Checking on the failures to see if they are more real than spurious.

**Comment by craigcomstock** [comment]
It does seem that both involved a debian 7 host which didn't reboot "properly". Maybe a changed IP address? Isn't that the running theory on these failures? @vpodzime were you working on adding UUID to DTs so we can get accurate IP addr info on reboots? Or will that help these errors?

**Comment by vpodzime** [comment]
> @vpodzime were you working on adding UUID to DTs so we can get accurate IP addr info on reboots? Or will that help these errors?

No, I only added the UUID handling to *mass.py* which spawns and destroys the VMs. So if VMs are now properly destroyed even if their IP address changes. However, deployment tests are designed to work with lists of IP addresses so passing objects instead would be a bigger change.

**Comment by craigcomstock** [comment]
Might want to take this one back to 3.12.x?

**Pull request merged**